test/stiefel.jl

```diff
@@ -296,4 +301,21 @@ include("utils.jl")
         q2 = retract(M, p, X, r2)
         @test is_manifold_point(M, q2)
     end
+
+    @testset "Canonical Metric" begin
+        M3 = MetricManifold(Stiefel(3, 2), CanonicalMetric())
+        p = [1.0 0.0; 0.0 1.0; 0.0 0.0]
+        X = [0.0 0.0; 0.0 0.0; 1.0 1.0]
+        q = exp(M3, p, X)
+        Y = [0.0 0.0; 0.0 0.0; -1.0 1.0]
+        r = exp(M3, p, Y)
+        @test isapprox(M3, p, log(M3, p, q), X)
+        @test isapprox(M3, p, log(M3, p, r), Y)
+        @test inner(M3, p, X, Y) == 0
+        a = 1.4
+        Z = [0.0 2a; -2a 0.0; -a/2 a/2]
+        s = exp(M3, p, Z)
+        Z2 = log(M3, p, s)
+        @test isapprox(M3, p, Z, Z2)
```

Comment:
It would be nice to test `distance` (we had (have?) a bug, where `distance` worked for Euclidean Stiefel but returned a wrong answer).

Reply:
I hope we had that bug since I changed the transparency in this PR for that.

I can add a test for sure.